### PR TITLE
Extract pure createSplitSource and drop visitScanAndFilter

### PR DESCRIPTION
**Pull request by lukasz-stec**
## Description
Make `visitScanAndFilter` function without side effects and rename it to more telling `createSplitSource`

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/sql/planner/SplitSourceFactory.java [hunk 1]
```diff
@@ -154,14 +156,15 @@ public Map<PlanNodeId, SplitSource> visitExplainAnalyze(ExplainAnalyzeNode node,
...
-            return visitScanAndFilter(node, Optional.empty());
+            SplitSource splitSource = createSplitSource(node.getTable(), node.getAssignments(), Optional.empty());
+
+            splitSources.add(splitSource);
+
+            return ImmutableMap.of(node.getId(), splitSource);
         }
 
-        private Map<PlanNodeId, SplitSource> visitScanAndFilter(TableScanNode node, Optional<FilterNode> filter)
+        private SplitSource createSplitSource(TableHandle table, Map<Symbol, ColumnHandle> columnHandles, Optional<Expression> filterPredicate)
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/sql/planner/SplitSourceFactory.java:166]: columnHandles -> assignments

(eg can `columnHandles.values()` have duplicates? what's the meaning of map keys?)

**Review by findepi — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/sql/planner/SplitSourceFactory.java (hunk 1 above)
Comment (lukasz-stec) [reply core/trino-main/src/main/java/io/trino/sql/planner/SplitSourceFactory.java]: renamed to `assignments`

File: core/trino-main/src/main/java/io/trino/sql/planner/SplitSourceFactory.java [hunk 2]
```diff
@@ -255,7 +254,11 @@ public Map<PlanNodeId, SplitSource> visitValues(ValuesNode node, Void context)
         public Map<PlanNodeId, SplitSource> visitFilter(FilterNode node, Void context)
         {
             if (node.getSource() instanceof TableScanNode scan) {
-                return visitScanAndFilter(scan, Optional.of(node));
+                SplitSource splitSource = createSplitSource(scan.getTable(), scan.getAssignments(), Optional.of(node.getPredicate()));
+
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/SplitSourceFactory.java:258]: remove empty line?

File: core/trino-main/src/main/java/io/trino/sql/planner/SplitSourceFactory.java [hunk 3]
```diff
@@ -154,14 +156,15 @@ public Map<PlanNodeId, SplitSource> visitExplainAnalyze(ExplainAnalyzeNode node,
         @Override
         public Map<PlanNodeId, SplitSource> visitTableScan(TableScanNode node, Void context)
         {
-            return visitScanAndFilter(node, Optional.empty());
+            SplitSource splitSource = createSplitSource(node.getTable(), node.getAssignments(), Optional.empty());
+
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/SplitSourceFactory.java:160]: nit: remove empty line

**Pull request merged**